actuators/machine.go

```diff
@@ -78,14 +79,25 @@ func (m *Machine) Create(ctx context.Context, c *clusterv1.Cluster, machine *clu
 	if setValue == clusterAPIControlPlaneSetLabel {
 		if len(controlPlanes) > 0 {
 			fmt.Println("Adding a control plane")
-			controlPlaneNode, err := actions.AddControlPlane(c.Name, machine.Spec.Versions.ControlPlane)
+			controlPlaneNode, err := actions.AddControlPlane(c.Name, machine.GetName(), machine.Spec.Versions.ControlPlane)
 			if err != nil {
 				fmt.Printf("%+v", err)
 				return err
 			}
-			name := providerID(controlPlaneNode.Name())
-			machine.Spec.ProviderID = &name
-			return m.save(old, machine)
+			nodeUID, err := actions.GetNodeRefUID(c.GetName(), controlPlaneNode.Name())
```

Comment:
Isn't this same code repeated below? maybe use a helper function? also will reduce the overall `create` function length.

Reply:
agreed, I'll see if I can extract this